ppm-main/src/main/java/com/ctu/se/oda/model11/api/TaskApi.java

```diff
@@ -66,4 +81,32 @@ public ResponseEntity<?> deleteTask(@PathVariable String taskId) {
         taskApplication.deleteTask(UUID.fromString(taskId));
         return new ResponseEntity<>(HttpStatus.OK);
     }
+
+    @PostMapping("/{taskId}/material/{materialId}")
+    public ResponseEntity<?> addMaterialToTask(@PathVariable String taskId,
+                                               @PathVariable String materialId,
+                                               @RequestParam Double amount) {
+        var retrieveTask = taskRepository.findById(UUID.fromString(taskId)).get();
```

Comment:
Only the infrastructure layer is allowed to calling repository and making changes to database. Please refactor code to forward logic to application layers and then infrastructure layer.